src/components/views/rooms/RoomBreadcrumbs.js

```diff
@@ -38,22 +39,22 @@ export default class RoomBreadcrumbs extends React.Component {
     componentWillMount() {
         this._dispatcherRef = dis.register(this.onAction);
 
-        const roomStr = localStorage.getItem("mx_breadcrumb_rooms");
-        if (roomStr) {
-            try {
-                const roomIds = JSON.parse(roomStr);
-                this.setState({
-                    rooms: roomIds.map((r) => {
-                        return {
-                            room: MatrixClientPeg.get().getRoom(r),
-                            animated: false,
-                        };
-                    }).filter((r) => r.room),
-                });
-            } catch (e) {
-                console.error("Failed to parse breadcrumbs:", e);
+        let storedRooms = SettingsStore.getValue("breadcrumb_rooms");
+        if (!storedRooms || !storedRooms.length) {
+            // Fallback to the rooms stored in localstorage for those who would have had this.
+            // TODO: Remove this after a bit - the feature was only on develop, so a few weeks should be plenty time.
```

Comment:
does it make sense to have this fallback code while it's still in labs? These things tend to linger for a long time ...

Reply:
Only for the people who have used the feature and may already rely on it (like myself). As part of pulling it out of labs, this can go.

Reply:
(I also have a giant note on my desk to remove it)